### PR TITLE
support `all_atlas_group_infos`

### PR DESCRIPTION
**Pull request by ThomasBreuer**
- added `all_atlas_group_infos`,
  analogous to the `all_something` functions from other group libraries
- added `atlas_group` with argument a dictionary returned by
  `all_atlas_group_infos`
- added `rank_action`
- generalized `translate_group_library_args` and `find_index_function`
  such that they fit to
  the use for character tables (via `_ctbl_filter_attrs`)
  and Atlas groups (via `_atlas_group_filter_attrs`)
- extended the tests accordingly

### CODE REVIEW

File: src/Groups/gsets.jl [hunk 1]
```diff
@@ -760,6 +761,38 @@ all_blocks(G::PermGroup) = Vector{Vector{Int}}(GAP.Globals.AllBlocks(G.X))
 #TODO: Do we really want to act on the set of moved points?
 
 
+"""
+    rank_action(G::PermGroup, L::AbstractVector{Int} = 1:degree(G))
+
+Return the rank of the action of `G` on `L`, i.e., the number of orbits
```
Comment (fingolfin) [inline src/Groups/gsets.jl]: Never saw that definition before. Perhaps provide a reference?

File: src/Groups/gsets.jl [hunk 2]
```diff
@@ -760,6 +761,38 @@ all_blocks(G::PermGroup) = Vector{Vector{Int}}(GAP.Globals.AllBlocks(G.X))
...
+of the point stabilizer.
```
Comment (fingolfin) [inline src/Groups/gsets.jl]: ... acting on what? I would have assumed after reading this that the point being stabilized is not included, but in your code it is... so the rank is 1 iff the group has degree 1 (and thus is trivial), and 2 iff the group acts 2-transitively?

File: src/Groups/libraries/atlasgroups.jl [hunk 3]
```diff
@@ -44,6 +49,164 @@ function atlas_group(::Type{T}, name::String) where T <: Union{PermGroup, Matrix
   return TT(G)
 end
 
+
+"""
+    atlas_group(info::Dict)
+
+Return the group from the Atlas of Group Representations
+that is defined by `info`.
+Typically, `info` is returned by [`all_atlas_group_infos`](@ref).
```
Comment (fingolfin) [inline src/Groups/libraries/atlasgroups.jl]: Maybe more like this?
```suggestion
Typically, `info` is obtained from [`all_atlas_group_infos`](@ref).
```

File: src/Groups/gsets.jl (hunk 2 above)
Comment (ThomasBreuer) [reply src/Groups/gsets.jl]: Yes.

**Review by fingolfin — APPROVED** [review]

**Pull request merged**